dnsapi/dns_fh.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env sh
+########################################################################
+# Futurehosting script for acme.sh
+#
+# Environment variables:
+#
+#  - FH_API_TOKEN  (your Futurehosting API Token)
+#  Note: If you do not have an API token, one can be generated at:
+#        https://my.futurehosting.com/api-token
+#
+# Author: Frank Laszlo <flaszlo@nexcess.net>
+
+FH_API_URL="https://my.futurehosting.com/"
+FH_API_VERSION="0"
+
+# dns_fh_add() - Add TXT record
+# Usage: dns_fh_add _acme-challenge.subdomain.domain.com "XyZ123..."
+dns_fh_add() {
+  host="${1}"
+  txtvalue="${2}"
+
+  _debug host "${host}"
+  _debug txtvalue "${txtvalue}"
+
+  if ! _check_fh_api_token; then
+    return 1
+  fi
+
+  _info "Using Futurehosting"
+  _debug "Calling: dns_fh_add() '${host}' '${txtvalue}'"
+
+  _debug "Detecting root zone"
+  if ! _get_root "${host}"; then
+    _err "Zone for domain does not exist."
+    return 1
+  fi
+  _debug _zone_id "${_zone_id}"
+  _debug _sub_domain "${_sub_domain}"
+  _debug _domain "${_domain}"
+
+  _post_data="{\"zone_id\": \"${_zone_id}\", \"type\": \"TXT\", \"host\": \"${host}\", \"target\": \"${txtvalue}\", \"ttl\": \"300\"}"
+
+  if _rest POST "dns-record" "${_post_data}" && [ -n "${response}" ]; then
+    _record_id=$(printf "%s\n" "${response}" | _egrep_o "\"record_id\":\s*[0-9]+" | cut -d : -f 2 | tr -d " " | _head_n 1)
```

Comment:
do not use `'\s'`,  it's not crossing platforms.
please just use plain space char `' '` instead.